### ISSUE TITLE
Working fine on local (MAMP) - not on live site.

### ISSUE THREAD

**Issue by eheiser** [codemonauts/craft-instagram-feed#4]
It's working fine on my local install (MAMP), on the live site all I get in web.log iis
```
2019-05-09 20:08:51 [-][1][-][error][codemonauts\instagramfeed\services\InstagramService::getInstagramData] Instagram profile data could not be fetched. Wrong account name or not a public profile.
2019-05-09 20:08:51 [-][1][-][info][application] $_GET = []
```
yet the account name is correct and it is a public profile. Any suggestions for further debugging?

**Comment by kringkaste** [comment]
Hi! We are using the file_get_contents function to fetch the data. Maybe the php setting "allow_url_fopen" is set to "1" on your MAMP and "0" on your production server? Can you check this?

**Comment by eheiser** [comment]
On the live server allow_url_fopen is set to on. PHP version 7.1.28

**Comment by iamtompickering** [comment]
I'm having the same issue. Have you had any luck with this?

Running PHP version 7.0 and I have just tried adding the option of `allow_url_fopen = 1` but it didn't resolve the issue.

**Comment by kringkaste** [comment]
Please test the new 1.0.3 release and make use of Guzzle instead of php's file stream. You can enable it in the config file.

**Comment by hendrikeng** [comment]
doesn't work for me either....tried allow_url_fopen = 1 as well as guzzle instad of php stream

**Comment by fleaz** [comment]
Hey everyone,

to make this easier to debug could you please enable debug logging in the config file (like described here: https://github.com/codemonauts/craft-instagram-feed/issues/21#issuecomment-541001390.

This will give us the actual response you got from IG.  There we can see if they e.g. changed the site structure or if your servers are on some of their blacklist.

Felix

**Comment by pj258** [comment]
Hi, 
same issue for me.  Working on local dev, not working on Live Site. 
Tried all stuff from above. 
Craft CMS 3.3.13 / + current .git version of the plugin

dump / log file
[1573044734.txt](https://github.com/codemonauts/craft-instagram-feed/files/3814554/1573044734.txt)

**Comment by fleaz** [comment]
Hey @pj258 
by looking at the response you got from IG, you can see that you got redirected to the login page instead of showing the profile page our plugin requested. This usually means that you triggered some kind of "bot detection" on their side. To our understanding this is bound to your IP address and you can't get it removed.

Sometimes it will start working again in the future by itself, but this is uncertain.

Sorry that I can't really help you, but the blacklists used by IG are pretty arbitrary and totally intransparent :/ 
We e.g. use this plugin ourself on some EC2 instances and they have been working fine for months.

Best regards,
Felix

**Comment by fleaz** [comment]
Hey,

because multiple of our customers who use this plugin faced the same problem of beeing blacklisted by Instagram and therefore couldn't use our plugin, we started to implement a new feature which will circumvent this and use a proxy server to get the data.
If you still have this problem and wan't to give our plugin another chance, we would love to have you in our beta test for the next version.

Just write a mail to support@codemonauts.com and we will give you a beta key and all the details on how to use it :)

Best regards,
Felix

**Comment by pierrestoffe** [comment]
To give a bit more context to this issue: I have found out that Instagram has started to force Instagram users to log in before they can see an Instagram post. 

The catch is that they are implementing this update region by region. For example, in Belgium where I am located I have no issue accessing Instagram posts without being logged in. In the Netherlands though, where most of my staging and prod servers are located, Instagram requires an authentication (hence the redirection to the login form) before one can access a post.

This means that unless Instagram changes their mind, everyone will soon face the same limitation and neither the unauthenticated scraping nor the `?__a=1` URL parameter trick will work anymore.

**Comment by fleaz** [comment]
> The catch is that they are implementing this update region by region. For example, in Belgium where I am located I have no issue accessing Instagram posts without being logged in. In the Netherlands though, where most of my staging and prod servers are located, Instagram requires an authentication (hence the redirection to the login form) before one can access a post.

This has (as far as we discovered) not really something to do with your region but what "kind" of IP address you use to talk to instagram.com. As you mentioned you used your private internet access at home (IP range from an ISP) and in the Netherlands you used your prod and staging servers (IP from a data center). 

They force you to login when they see that you talk to them from a server IP because "real" Instagram traffic never comes from a server and normally always comes from a private IP from an ISP (IG app on a smartphone or people browsing the website).  Therefore we run our proxy infrastructure behind multiple private internet connections  to simulate traffic from real people and circumvent their blocking.

Of course they could theoretically also force all humans to login in to see pictures but imho they won't do that. They will always show you a bit of content to get you curious and then show you the "Get an account to see more" popup just like they are doing it on e.g. on Facebook

**Comment by pierrestoffe** [comment]
That's an interesting take and I think you're right actually. I just tried accessing an Instagram post using a Belgian VPN connection and I am blocked by the Instagram login form. I'm not blocked when accessing from my normal Belgian connection.

Looking forward to test your solution, I'll send you an email.
